packages/terra-status-view/src/StatusView.jsx

```diff
@@ -36,223 +43,146 @@ const propTypes = {
-    const titleNodeHeight = this.titleNode.getBoundingClientRect().height;
-    const actionNodeHeight = this.actionsNode ? this.actionsNode.getBoundingClientRect().height : 0;
-    const messageNodeHeight = this.messageNode ? this.messageNode.getBoundingClientRect().height : 0;
-    const dividerNodeHeight = this.dividerNode ? this.dividerNode.getBoundingClientRect().height : 0;
-    contentHeight = (titleNodeHeight + actionNodeHeight + messageNodeHeight + dividerNodeHeight);
-
-    contentWithGlyphHeight = contentHeight + (this.glyphNode ? this.glyphNode.getBoundingClientRect().height : 0);
-    // if inner view node height is lesser than the components along with glyph inside the inner view.
-    // and the height difference is very less (i.e 0.0001) then add that comparison value to the
-    // the inner view height.
-    if (contentWithGlyphHeight - viewHeight <= 0.0001) {
-      newViewHeight += 0.0001;
-    }
-
-    if (this.glyphNode && newViewHeight >= contentWithGlyphHeight && this.innerNode.offsetWidth >= this.glyphNode.offsetWidth) {
-      // if glyph exists and can fit inside the container's height and width, distribute remaining padding to center
-      showGlyph = true;
-      if (!this.props.isAlignedTop) {
-        const remainingHeight = (viewHeight > contentWithGlyphHeight) ? (viewHeight - contentWithGlyphHeight) : 0;
-        paddingTop = remainingHeight * 0.4;
-        paddingBottom = remainingHeight * 0.6;
-      }
-    } else if (!this.props.isAlignedTop) {
-      // glyph does not exist or does not fit, distribute remaining padding if any to center
-      const remainingHeight = viewHeight - contentHeight;
-      if (remainingHeight > 0) {
-        paddingTop = remainingHeight * 0.4;
-        paddingBottom = remainingHeight * 0.6;
-      }
-    }
-
-    // set which components can fit into the view
-    this.setState({
-      showGlyph,
-      paddingTop,
-      paddingBottom,
-    });
-  }
-
-  render() {
-    const {
-      children,
-      customGlyph,
-      isAlignedTop,
-      isGlyphHidden,
-      message,
-      title,
-      variant,
-      ...customProps
-    } = this.props;
-
-    const { intl } = this.context;
-
-    let glyphSection;
-    if (!isGlyphHidden && this.state.showGlyph) {
-      if (variant === StatusViewVariants.CUSTOM) {
-        glyphSection = (
-          <div className={cx('glyph')} ref={(element) => { this.glyphNode = element; }}>
-            {customGlyph}
-          </div>
-        );
-      } else {
-        glyphSection = (
-          <div className={cx('glyph')} ref={(element) => { this.glyphNode = element; }}>
-            <svg className={cx(variant)} />
-          </div>
-        );
-      }
-    }
+  const generatedButtonsArray = [];
+  buttonAttrsArray.forEach(button => generatedButtonsArray.push(
+    <Button {...button} className={cx(['button', button.className])} />,
+  ));
+  return generatedButtonsArray;
```

Comment:
Wanna use map?
```jsx
return buttonAttrsArray.map(button => <Button {...button} className={cx(['button', button.className])} />);
```

Reply:
Are key's required here? We should we document that if they are.

Reply:
updated and added a comment on the prop comment